### PR TITLE
Added doxygen configuration

### LINKED ISSUE #30: Code is missing documentation
Add code documentation via doxygen
 - automatically generate docu based on annotations in code

### LINKED ISSUE #42: Documentation should be autogenerated
The current doxygen documentation, as well as the diagrams are currently manually automated and pushed to the main repository. This makes it harder to read, and adds more workflow for devs. A way to alleviate this issue is to automatically generate the doxygen documentation on push and deploy it as a static webpage (e.g. on gh pages)

### LINKED ISSUE #44: Add git hook to build doxygen documentation
Currently users need doxygen plus extra tools installed to be able to build the documentation client side, this process could be made easier by providing a git hook to build the documentation via docker post-commit and post-receive. Optionally it should be possible to generate external and internal documentation without changing the config file.

### PR DESCRIPTION
**Pull request by Nanogamer7**
This PR adds a configured Doxyfile, including a workflow to automatically deploy via GitHub pages, and styling for the generated documentation. However due to the extend of undocumented code in the code base, this PR doesn't add complete documentation, which should be done after restructuring the code as described in #29. Some parts are documented as a sample on how future documentation could or should look like.

### CODE REVIEW

**Comment by Nanogamer7** [comment]
Documentation doesn't differentiate between internal documentation (i.e. documentation only relevant for further engine development) and external documentation (i.e. documentation of functionality included in the API, potentially also relevant for game developers)

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "debug",
+            "type": "cppdbg",
+            "miDebuggerPath": "c:\\mingw64\\bin\\gdb.exe",
```
Comment (Nimac0) [inline .vscode/launch.json:10]: might not be universally correct

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+    // See https://go.microsoft.com/fwlink/?LinkId=733558
+    // for the documentation about the tasks.json format
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "type": "shell",
+            "label": "build-engine-debug",
+            "command": "cd build; cmake -DCMAKE_BUILD_TYPE=Debug; cmake --build .",
```
Comment (Nimac0) [inline .vscode/tasks.json:9]: only works on windows (bc of ;)

File: .vscode/tasks.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            "problemMatcher": [],
+            "group": "build"
+        },
+        {
+            "type": "shell",
+            "label": "build-engine",
+            "command": "cd build; cmake -DCMAKE_BUILD_TYPE=Release; cmake --build .",
```
Comment (Nimac0) [inline .vscode/tasks.json:16]: see above

File: .vscode/tasks.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    },
+                ]
+            },
+            "problemMatcher": [],
+        },
+        {
+            "type": "process",
+            "label": "open-doxygen",
+            "command": "explorer",    
+            "windows": {
```
Comment (Nimac0) [inline .vscode/tasks.json:59]: potentially make functional on linux

File: engine.sublime-project [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    },
+	"build_systems": [
+		{
+			"name": "Build engine",
+			"shell_cmd": "cmake --build build",
+			"working_dir": "$project_path",
+			"variants": [
+				{
+					"name": "Debug",
+					"shell_cmd": "cmake -DCMAKE_BUILD_TYPE=Debug build && cmake --build build",
```
Comment (Nimac0) [inline engine.sublime-project:20]: only works on linux

**Review by Nimac0 — APPROVED** [review]
gud

**Pull request merged**